test/hotspot/jtreg/compiler/vectorization/runner/VectorizationTestRunner.java

```diff
@@ -131,7 +138,10 @@ private void runTestOnMethod(Method method) throws InterruptedException {
         WB.enqueueMethodForCompilation(method, COMP_LEVEL_C2);
         while (WB.getMethodCompilationLevel(method) != COMP_LEVEL_C2) {
             if (System.currentTimeMillis() - enqueueTime > COMP_THRES_SECONDS * 1000) {
-                fail("Method is not compiled after " + COMP_THRES_SECONDS + "s.");
+                // C2 compilation may timeout with extra compiler control options.
+                // We skip the correctness check in this case.
+                System.out.println("WARNING: Correctness check is skipped because C2 compilation times out");
+                return;
```

Comment:
Why not run with `BackgroundCompilation` OFF? The code will return from `WHITE_BOX.enqueueMethodForCompilation()` only after finishing compiling or compiler skip.
Compilation speed is very different on different platform. Using time for check is not reliable.

Reply:
> Why not run with BackgroundCompilation OFF?

I just explained the reason in my another reply above. Using time for check is to avoid jtreg timeout with extra VM options like `-XX:CompileThreshold=100 -XX:-TieredCompilation`. But I think this can also be avoided if we get reference result from C1 execution. What do you think of the new approach?